lib/active_record/virtual_attributes/virtual_fields.rb

```diff
@@ -69,23 +69,24 @@ def include_to_hash(value)
           when String, Symbol
             {value => {}}
           when Array
-            value.flatten.each_with_object({}) { |k, h| h[k] = {} }
+            value.flatten.each_with_object({}) do |k, h|
+              merge_includes(h, k)
+            end
```

Comment:
If we tread the string/symbol case as a special case, it would avoid converting `h` to a hash `h => {}` and avoid the deep_merge() call. But I think what we have is better overall

```suggestion
            value.flatten.each_with_object({}) do |k, h|
              if k.kind_of?(Symbol) || k.kind_of?(String)
                h[k] ||= {}
              else
                merge_includes(h, k)
              end
            end
```

Reply:
This case is pretty common.

I added it to the code so it wasn't in the back of my mind.

Reply:
I implemented this behavior.
If I cared enough about this "regression" to document it here, then I cared enough about it to write up some tests and fix it.